src/parser/mod.rs

```diff
@@ -3939,6 +3954,26 @@ impl<'a> Parser<'a> {
             })
     }
 
+    /// Return nth previous token, possibly whitespace
+    /// (or [`Token::EOF`] when before the beginning of the stream).
+    pub fn peek_prev_nth_token_no_skip(&self, n: usize) -> TokenWithSpan {
```

Comment:
```suggestion
    pub(crate) fn peek_prev_nth_token_no_skip(&self, n: usize) -> &TokenWithSpan {
```
we can probably start with the API being non-public. Then can we switch to returning a reference to avoid the mandatory cloning?

Reply:
Done 👍